components/runtime/wine/Makefile

```diff
@@ -0,0 +1,89 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL)". You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2019 Evgeny Litvinenko
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		wine
+COMPONENT_VERSION=	4.4
+COMPONENT_REVISION=     1
+COMPONENT_PROJECT_URL=	https://www.winehq.org/
+COMPONENT_SUMMARY=	Wine - Run Windows applications.
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.xz
+COMPONENT_ARCHIVE_HASH=	\
+    sha256:db3a09cf4b0b4329ffed8014073c9868b5ecb4826adfb4a91aa49bd2841207b0
+COMPONENT_DOWNLOAD_URL= git://source.winehq.org/git/$(COMPONENT_NAME).git
+COMPONENT_ARCHIVE_URL=	https://dl.winehq.org/wine/source/4.x/$(COMPONENT_ARCHIVE)
+COMPONENT_SIG_URL=	$(COMPONENT_ARCHIVE_URL).sign
+COMPONENT_FMRI=		runtime/wine
+COMPONENT_LICENSE=	LGPLv2.1
+COMPONENT_LICENSE_FILE=	$(COMPONENT_NAME).license
+COMPONENT_CLASSIFICATION=	Applications/System Utilities
+COMPONENT_BUGDB=	utility/wine
+
+USERLAND_GIT_REMOTE=	https://github.com/OpenIndiana/oi-userland.git
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/configure.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+gcc_OPT= -O3
+
+PATH= $(PATH.gnu)
+
+CONFIGURE_OPTIONS += LD="/usr/bin/ld"
+CONFIGURE_OPTIONS += AS="/usr/bin/gas"
+
+CONFIGURE_OPTIONS += --without-pulse --with-oss
```

Comment:
Why do we build without pulseaudio?

Reply:
performance most likely
better to use OSADA or OSS directly than go through pulse

Reply:
Most of the OpenIndiana desktop runs thru PulseAudio. I've heard of various issues with PulseAudio but never of performance. Are there performance issues with PulseAudio with wine?

Reply:
If it works with pulse, I think it should be built with it.

Reply:
ah I just tested https://mumble.info with the `winepulse.drv` driver, thanks to #4922/OI CR9732, the pulseaudio driver picks my GPU HD Audio card, _which doesn't even have an input pin to speak of_
using OSS I can pick the _real_ HD Audio card which has my mic and earphones plugged into the front panel jacks

Reply:
ok so pulseaudio on the native side is properly configured, and allows me to use the internal HD Audio sound card at `/dev/sound/4`.
The PulseAudio sink is wired up to `/dev/dsp4` because if both source and sink are wired to the same backend (OSS `/dev/dsp` or OSADA `/dev/sound/*`), the sound server jams up. (In fact, OSS _input_ simply does not seem to work at all, while OSADA jams up if configured to do both input and output at the same time.)
Now, all of this works fine natively once the above is considered. Once you step into wine, only output seems to work at all.